### PR TITLE
Enable `karmada-aggregated-apiserver` installation by `kubectl-karmada init`

### PR DESCRIPTION
**Pull request by prodanlabs**
**What type of PR is this?**
/kind feature
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1122

**Special notes for your reviewer**:
add examples
```
root@dev-k8s-master01:~# ./kubectl-karmada init -h
Installation options.

Usage:
  karmada init [flags]

Examples:

# This master IP is any node IP of kubernetes.
kubectl karmada init --master=192.168.1.2

# In the Intranet, CRDs resources specify local files.
kubectl karmada init --master=192.168.1.2 --crd /root/crds.tar.gz

# Private registry can be specified for all images.
kubectl karmada init --master=192.168.1.2 --etcd-image local.registry.com/library/etcd:3.5.1-0

# Deploy highly available(HA) karmada.
kubectl karmada init --master=192.168.1.2,192.168.1.3,192.168.1.4 --karmada-apiserver-replicas 3 --etcd-replicas 3

# Karmada uses external load balancing or HAip, and karmada certificate needs to join the IP.
kubectl karmada init --master=192.168.1.2,192.168.1.3,192.168.1.4 --cert-external-ip 10.235.1.2
```

add  deploy `karmada-aggregated-apiserver`, It seems that the features of  `karmada-aggregated-apiserver`  have not been realized?
```
root@dev-k8s-master01:~# kubectl -n karmada-system get pod
NAME                                                      READY   STATUS    RESTARTS   AGE
etcd-0                                                    1/1     Running   0          7m58s
karmada-aggregated-apiserver-6b46dcb9c6-ft6sf             1/1     Running   0          7m11s
karmada-apiserver-6b98df45df-k979g                        1/1     Running   0          7m55s
karmada-controller-manager-556cf896bc-qdwx2               1/1     Running   0          7m17s
karmada-scheduler-7b9d8b5764-rtg6x                        1/1     Running   0          7m20s
karmada-scheduler-estimator-kubernetes-65d9b768f8-wxbd7   1/1     Running   0          3m21s
karmada-webhook-7cf7986866-xk7qq                          1/1     Running   0          7m14s
kube-controller-manager-85c789dcfc-kbfps                  1/1     Running   0          7m23s
root@dev-k8s-master01:~# 
root@dev-k8s-master01:~# kubectl  --kubeconfig /etc/karmada/karmada-apiserver.config get clusters 
NAME         CREATED AT
kubernetes   2021-12-21T16:44:08Z
root@dev-k8s-master01:~# 
root@dev-k8s-master01:~# curl  --cacert /etc/karmada/ca.crt  --cert /etc/karmada/karmada.crt  --key /etc/karmada/karmada.key  -X GET "https://172.31.6.145:5443/apis/cluster.karmada.io/v1alpha1/cluster/kubernetes/proxy/api/v1/nodes"
{{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "the server could not find the requested resource",
  "reason": "NotFound",
  "details": {
    
  },
  "code": 404
}
root@dev-k8s-master01:~# kubectl  --kubeconfig /etc/karmada/karmada-apiserver.config get  apiservices.apiregistration.k8s.io  v1alpha1.cluster.karmada.io 
NAME                          SERVICE                                       AVAILABLE   AGE
v1alpha1.cluster.karmada.io   karmada-system/karmada-aggregated-apiserver   True        8m25s
```
**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
> add deploy karmada-aggregated-apiserver, It seems that the features of karmada-aggregated-apiserver have not been realized?

Yeah, the `karmada-aggregated-apiserver` now only registers the API, the proxy functionality is on the way.

File: pkg/karmadactl/cmdinit/karmada/deploy.go [hunk 1]
```diff
@@ -154,3 +162,61 @@ func getName(str, start, end string) string {
...
 	return str
 }
+
+func apiServiceResourceVersion(c *aggregator.Clientset, name string) string {
+	apiService, _ := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
+	return apiService.ResourceVersion
+}
+
+func initAPIService(clientSet *kubernetes.Clientset, restConfig *rest.Config) error {
+	// https://github.com/prodanlabs/karmada/blob/master/artifacts/deploy/apiservice.yaml
```
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/karmada/deploy.go:172]: I guess you forgot to remove it.

File: pkg/karmadactl/cmdinit/kubernetes/deploy.go [hunk 2]
```diff
@@ -310,11 +286,22 @@ func (i *InstallOptions) initComponent() {
...
-
 	if _, err := i.KubeClientSet.AppsV1().Deployments(i.Namespace).Create(context.TODO(), i.makeKarmadaWebhookDeployment(), metav1.CreateOptions{}); err != nil {
 		klog.Warning(err)
 	}
-	if err := WaitPodReady(i.KubeClientSet, i.Namespace, utils.MapToString(webhookLabels), 30); err != nil {
+	if err := WaitPodReady(i.KubeClientSet, i.Namespace, utils.MapToString(webhookLabels), waitPodReadyTimeout); err != nil {
+		klog.Warning(err)
+	}
+	// Create karmada-aggregated-apiserver
+	// https://github.com/prodanlabs/karmada/blob/master/artifacts/deploy/karmada-aggregated-apiserver.yaml
```
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/kubernetes/deploy.go:296]: ditto

**Comment by RainbowMango** [comment]
/retitle Enable `karmada-aggregated-apiserver` installation by `kubectl-karmada init`

File: pkg/karmadactl/cmdinit/kubernetes/deploy.go (hunk 2 above)
Comment (prodanlabs) [reply pkg/karmadactl/cmdinit/kubernetes/deploy.go:296]: The copied URL address is wrong. I'll change it.

**Pull request merged**